support/cas-server-support-pac4j-core/src/test/java/org/apereo/cas/validation/DelegatedAuthenticationAccessStrategyHelperTests.java

```diff
@@ -0,0 +1,108 @@
+package org.apereo.cas.validation;
+
+
+import org.apereo.cas.audit.AuditableExecution;
+import org.apereo.cas.audit.AuditableExecutionResult;
+import org.apereo.cas.authentication.principal.Service;
+import org.apereo.cas.services.RegisteredService;
+import org.apereo.cas.services.RegisteredServiceAccessStrategy;
+import org.apereo.cas.services.ServicesManager;
+
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import javax.servlet.http.HttpServletRequest;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests for {@link DelegatedAuthenticationAccessStrategyHelper}.
+ *
+ * @author Stefan Reiterer
+ * @since 6.5
+ */
+@Tag("Delegation")
+public class DelegatedAuthenticationAccessStrategyHelperTests {
+
+    @Test
+    public void isDelegatedClientAuthorizedForServiceIsNull() {
+        val serviceManager = mock(ServicesManager.class);
+        val delegatedAuthenticationPolicyEnforcer = mock(AuditableExecution.class);
+
+        DelegatedAuthenticationAccessStrategyHelper delegatedAuthenticationAccessStrategyHelper =
+                new DelegatedAuthenticationAccessStrategyHelper(serviceManager, delegatedAuthenticationPolicyEnforcer);
+
+        val httpServletRequest = mock(HttpServletRequest.class);
+
+        /*
+        Without service delegated authentication policy cannot be evaluated and so Delegated Auth. should not be
+        authorized for the given client-name
+         */
```

Comment:
The standard in CAS is to avoid comments in methods in favor of method or class level comments that would show up in the javadocs.